test/csi/toolkit/counter_tool.cpp

```diff
@@ -1,28 +1,27 @@
+#include "csi.h"
 #include <stdio.h>
 #include <stdlib.h>
 
+extern "C" {
 long times_accessed_by_size[4];
 
-static void __csi_destroy() {
+WEAK void __csi_destroy() {
```

Comment:
`__csi_destroy` isn't a CSI function; it's just called by `atexit` to do the printing. I'd keep it static.